tests/test_ilastik/test_applets/test_serverConfiguration/test_configStore.py

```diff
@@ -1,162 +1,68 @@
-        assert "127.0.0.1" == srv.address
-        assert "5543" == srv.port
-
-    def test_parsing_multiserver(self, parse):
-        servers = parse(self.MALFORMED_MULTI_SERVER)
-
-        assert 2 == len(servers)
-        fst, snd = servers
+    def get(self, group, key, default=None):
+        return self.config[group].get(key, default)
 
-        assert isinstance(fst, types.ServerConfig)
-        assert "MyServer1" == fst.name
-        assert "local" == fst.type
-        assert "127.0.0.1" == fst.address
-        assert fst.autostart
-        assert "5543" == fst.port
 
-        assert isinstance(snd, types.ServerConfig)
-        assert "MyServer3" == snd.name
-        assert "remote" == snd.type
-        assert "8.8.8.8" == snd.address
-        assert "5543" == snd.port
-        assert "testuser" == snd.username
-        assert not snd.autostart
-        assert "/home/user/.ssh/id_rsa" == snd.ssh_key
-
-    def test_parsing_noserver(self, parse):
-        servers = parse(self.NO_SERVER)
-        assert not servers
+@pytest.fixture
+def prefs_stub():
+    return PreferencesStub()
 
 
-class TestServerParseDevices:
-    NORMAL_DEVS = """
-[tiktorch-server::myid1]
-    name = MyServer1
-    type = local
-    address = 127.0.0.1
-    port = 5543
-    devices =
-        cpu0::CPU0
-        gpu1::MyCoolGPU::enabled
-"""
-
-    MALFORMED_DEVS = """
-[tiktorch-server::gpu]
-    name = MalformedGPU
-    type = local
-    address = 127.0.0.1
-    port = 5543
-    devices =
-        ::cpu0::CPU0:::::::
-        gpu1::MyCoolGPU::1
-
-
-        gpu2::NormalGPU::enabled
-"""
-
-    EMPTY_DEVS = """
-[tiktorch-server::gpu2]
-    name = MalformedGPU2
-    type = local
-    address = 127.0.0.1
-    port = 5543
-    devices =
-"""
-
-    NO_DEVS = """
-[tiktorch-server::gpu2]
-    name = MalformedGPU2
-    type = local
-    address = 127.0.0.1
-    port = 5543
-"""
+@pytest.fixture
+def config_storage(prefs_stub):
+    return ServerConfigStorage(prefs_stub)
 
-    @pytest.fixture
-    def parse(self):
-        def _parse(conf_str):
-            conf = ConfigParser()
-            conf.read_string(conf_str)
-            srv_storage = ServerConfigStorage(conf, dst='')
-            servers = srv_storage.get_servers()
-            assert len(servers) == 1
-            return servers[0].devices
 
-        return _parse
+class TestServerParse:
+    def test_parsing_server(self, prefs_stub, config_storage):
+        prefs_stub.set_server_configs(
+            [
+                {
+                    "id": "myid1",
+                    "name": "MyServer1",
+                    "address": "127.0.0.1:4412",
+                    "devices": [
+                        {"id": "cpu0", "name": "CPU0", "enabled": False},
+                        {"id": "gpu1", "name": "MyCoolGPU", "enabled": True},
+                    ],
+                }
+            ]
+        )
+        servers = config_storage.get_servers()
+        assert 1 == len(servers)
```

Comment:
i know of course order here doesn't matter, but for me, logically reading from left to right `len(servers)` should be first... but you seem to prefer it the other way round, which is also fine.